### PR TITLE
Min count validation for repeating groups

### PR DESCRIPTION
**Pull request by Magnusrm**
## Description
Adding minCount as a layout configuration for repeating groups and implementing minItems from schema validations.

Validating a minimum number of rows is now possible through setting the `minCount` property in the formLauyout for the selected group component. Setting a validation this way ensures that the validation will respect `hiddenRow` functionality which hides rows in the frontend from the user, but still has the data stored in the datamodel.

You can also set `minItems` in the json.schema for the datamodel, on the list that contains entries in your repeating group, which will work as a validation for a minimum number of rows. This will however not take `hiddenRows` into consideration.

## Related Issue(s)

- #1098 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [x] Has been added/updated
  https://github.com/Altinn/altinn-studio-docs/pull/992
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Changes/additions to component properties
  - [x] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [ ] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by olemartinorg** [comment]
Notes to self: Interesting results in percy! I think I'll need to fix the datepicker testing in Cypress or add a `.no-visual-testing` class to it so that we don't end up with diffs every time the dates change.. Also, I think I should sort the list of attachments in a Summary by name, so that we don't get diffs when these are out of order.

File: src/layout/Group/types.d.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export interface IGroupEditProperties {
 export interface ILayoutGroup extends ILayoutCompBase<'Group'> {
   children: string[];
   maxCount?: number;
+  minCount?: number;
```
Comment (olemartinorg) [inline src/layout/Group/types.d.ts:32]: You should add this to `layout.schema.v1.json` as well. 🙏

File: src/utils/validation/validation.ts [hunk 2]
```diff
@@ -1338,6 +1351,36 @@ export function validateGroup(groupId: string, state: IRuntimeState, onlyInRowIn
...
+  const repeatingGroupMinCountValid = repeatingGroupMinCount <= repeatingGroupVisibleRows;
+
+  // if not valid, return appropriate error message
+  if (!repeatingGroupMinCountValid) {
+    const errorMessage = getParsedLanguageFromKey(
+      'validation_errors.minItems',
+      language,
+      [repeatingGroupMinCount],
+      true,
+    );
```
Comment (olemartinorg) [inline src/utils/validation/validation.ts:1377]: This needs refactoring to `useLanguage`/`langTools` if you merge from main now

File: src/utils/validation/validation.ts [hunk 3]
```diff
@@ -1338,6 +1351,36 @@ export function validateGroup(groupId: string, state: IRuntimeState, onlyInRowIn
   );
 }
 
+/*
+ * Validates a repeating groups component specific validations
+ * @param node the node to validate
+ * @param language the current language
+ */
+export function validateRepeatingGroup(node, language: ILanguage): IComponentValidations {
```
Comment (olemartinorg) [inline src/utils/validation/validation.ts:1359]: Make sure you coordinate with @bjosttveit if he's currently moving these into the component `def` classes

**Review by olemartinorg — APPROVED** [review]
You should merge from main once more to clean up the language functions, and add the `minCount` to the schema. 😊 Looks good otherwise! 🥳

File: src/layout/Group/types.d.ts (hunk 1 above)
Comment (Magnusrm) [reply src/layout/Group/types.d.ts:32]: Definetely! Oversight on my part 🤓

File: src/utils/validation/validation.ts (hunk 2 above)
Comment (Magnusrm) [reply src/utils/validation/validation.ts:1377]: Will do! 👍

**Pull request merged**